src/main/java/net/minecraftforge/datafix/ForgeDataFixerCustomFixersModStateTransition.java

```diff
@@ -0,0 +1,215 @@
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * Event handler utility class for managing the events related to custom DFU fixers.
+ * This particular class is used to handle firing of the parallel mod loading events
+ * for the DFU Fixer setup.
+ *
+ * These events are fired immediately after mod construction and before the rest of the game
+ * loads, even before registries are fired.
+ *
+ * THIS CLASS IS FOR INTERNAL USE ONLY!
+ */
+public class ForgeDataFixerCustomFixersModStateTransition
+{
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    /**
+     * Builds a new mod state transition which can be used to fire the appropriate fixer registration events on the mod bus.
+     *
+     * @return The mod state transition used to add DFU fixers.
+     */
+    public static IModStateTransition get()
+    {
+        return new ForgeSerialModStateTransition<>(
+          ForgeDataFixerCustomFixersModStateTransition::generateFixersEvents,
+          ForgeDataFixerCustomFixersModStateTransition::preDispatchFixerEvents,
+          ForgeDataFixerCustomFixersModStateTransition::postDispatchFixerEvents,
+          ForgeDataFixerCustomFixersModStateTransition::rebuildRulesAfterSchemaEvents
+        );
+    }
+
+    /**
+     * Gives access to the list of schemas registered with the data fixers.
+     * If we are currently running an incompatible version of DFU or somebody messed with it,
+     * then an empty list is returned.
+     *
+     * @return The list of all known schemas to handle.
+     */
+    private static List<ForgeSchema> getAllSchemas()
+    {
+        //Check if we are running in a compatible environment.
+        if (DataFixers.getDataFixer() instanceof ForgeDataFixerDelegateHandler forgeDataFixerDelegateHandler)
+        {
+            //Forge DFU wrapper is active, grab its configured schemas.
+            return forgeDataFixerDelegateHandler.getAllSchemas();
+        }
+
+        //Incompatible environment, return an empty list.
+        return Collections.emptyList();
+    }
+
+    /**
+     * Builder function for creating the events that are fired on start up.
+     * This returns one generator per schema, and keeps the schema in the order defined.
+     *
+     * The returned generator creates one event internally per schema and mod container, and then stores them for reuse.
+     *
+     * @return The stream of event generators used to fire one event per mod per schema.
+     */
+    private static Stream<IModStateTransition.EventGenerator<?>> generateFixersEvents()
+    {
+        //Grab the schemas
+        List<ForgeSchema> schemas = Lists.newArrayList(getAllSchemas());
+        //Sort them based on their version key.
+        schemas.sort(Comparator.comparingInt(Schema::getVersionKey));
+
+        //Create one generator per schema.
+        return schemas.stream().map(InternalEventGenerator::new);
```

Comment:
Given this method's use of streams I feel like you might as well keep it entirely using streams and just do
```java
return getAllSchemas().stream()
              .sorted(Comparator.comparingInt(Schema::getVersionKey))
              .map(InternalEventGenerator::new);
```